lib/rucio/client/uploadclient.py

```diff
@@ -686,15 +686,15 @@ def _retry_protocol_stat(self, protocol, pfn):
                 # The stat succeeded here, but the checksum failed
                 raise error
             except Exception as error:
-                self.logger.debug('stat: unexpected error=%s' % error)
+                self.logger(logging.DEBUG, 'stat: unexpected error=%s' % error)
                 fail_str = ['The requested service is not available at the moment', 'Permission refused']
                 if any(x in str(error) for x in fail_str):
                     raise error
-                self.logger.debug('stat: unknown edge case, retrying in %ss' % 2**attempt)
+                self.logger(logging.DEBUG, 'stat: unknown edge case, retrying in %ss' % 2**attempt)
                 time.sleep(2**attempt)
         return protocol.stat(pfn)
 
-    def _create_protocol(self, rse_settings, operation, force_scheme=None):
+    def _create_protocol(self, rse_settings, operation, force_scheme=None, logger=logging.log):
```

Comment:
be consistent in this function as well. I would suggest using `self.logger` if it is available, instead of a logger parameter.